src/app/client/src/app/modules/shared-feature/components/collection-player/collection-player.component.ts

```diff
@@ -601,6 +604,23 @@ export class CollectionPlayerComponent implements OnInit, OnDestroy, AfterViewIn
     });
   }
 
+  getOnlyRequiredProperties(collection) {
```

Comment:
Take 'requiredKeys' also as an input to this method. So that anybody can use this method to parse the collection.

Write this method in utils & call that from 
`
`
const requiredProps = ['mimeType', 'visibility', 'identifier', 'selected', 'name', 'contentType', 'children',
    'primaryCategory', 'additionalCategory', 'parent'];
collection = utils.reduceTreeProps(collection, requiredProps);
`